iohub/dataset_base.py

```diff
@@ -0,0 +1,71 @@
+from abc import ABC, abstractmethod
+from typing import Generator, Union
+
+import numpy as np
+from numpy.typing import ArrayLike
+
+
+class BaseFOV(ABC):
+    # NOTE: not using the `data` method from ngff Position
+
+    @property
+    @abstractmethod
+    def channels(self) -> list[str]:
+        raise NotImplementedError
+
+    def channel_index(self, key: str) -> int:
+        """Return index of given channel."""
+        return self.channels.index(key)
+
+    @property
+    @abstractmethod
+    def shape(self) -> tuple[int, int, int, int, int]:
+        # NOTE: suggestion, fix dimension to 5?
+        # We could me more restrictive than ome-zarr
+        raise NotImplementedError
+
+    @abstractmethod
+    def __getitem__(
+        self,
+        key: Union[int, slice, tuple[Union[int, slice], ...]],
+    ) -> ArrayLike:
+        """
+        Output object must support __array__ interface, np.asarray(...).
+        """
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def ndim(self) -> int:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def dtype(self) -> np.dtype:
+        raise NotImplementedError
+
+    @property
+    def scale(self) -> list[float]:
```

Comment:
Should there be a constraint on the length? It's a tricky thing since NGFF has scale in channel dimension, which has no meaning and should always be 1. The time 'scale' is a bit ambiguous too...

Reply:
We can also just do `spatial_scale` here and it will always be a 3-tuple.

Reply:
I'll do `zyx_scale`, it's more informative, IMO.